controllers/zync_controller.go

```diff
@@ -72,76 +73,27 @@ func (r *ZyncReconciler) Reconcile(ctx context.Context, req ctrl.Request) (ctrl.
 		instance.Spec,
 	)
 
-	// Calculate rollout triggers
-	triggers, err := r.TriggersFromSecretDefs(ctx,
-		gen.ZyncSecretDefinition(),
-	)
+	// Shared resources
+	resources := gen.Resources()
+
+	// Api resources
+	var api_resources []basereconciler.Resource
+	api_resources, err = r.NewDeploymentWorkloadWithTraffic(ctx, instance, r.GetScheme(), &gen.API, &gen.API)
 	if err != nil {
-		return ctrl.Result{}, err
+		return r.ManageError(ctx, instance, err)
 	}
+	resources = append(resources, api_resources...)
 
-	err = r.ReconcileOwnedResources(ctx, instance, basereconciler.ControlledResources{
-		Deployments: []basereconciler.Deployment{
-			{
-				Template:        gen.API.Deployment(),
-				HasHPA:          !instance.Spec.API.HPA.IsDeactivated(),
-				RolloutTriggers: triggers,
-			},
-			{
-				Template:        gen.Que.Deployment(),
-				HasHPA:          !instance.Spec.Que.HPA.IsDeactivated(),
-				RolloutTriggers: triggers,
-			},
-		},
-		SecretDefinitions: []basereconciler.SecretDefinition{
-			{
-				Template: gen.ZyncSecretDefinition(),
-				Enabled:  true,
-			},
-		},
-		Services: []basereconciler.Service{
-			{
-				Template: gen.API.Service(),
-				Enabled:  true,
-			}},
-		PodDisruptionBudgets: []basereconciler.PodDisruptionBudget{
-			{
-				Template: gen.API.PDB(),
-				Enabled:  !instance.Spec.API.PDB.IsDeactivated(),
-			},
-			{
-				Template: gen.Que.PDB(),
-				Enabled:  !instance.Spec.Que.PDB.IsDeactivated(),
-			},
-		},
-		HorizontalPodAutoscalers: []basereconciler.HorizontalPodAutoscaler{
-			{
-				Template: gen.API.HPA(),
-				Enabled:  !instance.Spec.API.HPA.IsDeactivated(),
-			},
-			{
-				Template: gen.Que.HPA(),
-				Enabled:  !instance.Spec.Que.HPA.IsDeactivated(),
-			},
-		},
-		PodMonitors: []basereconciler.PodMonitor{
-			{
-				Template: gen.API.PodMonitor(),
-				Enabled:  true,
-			},
-			{
-				Template: gen.Que.PodMonitor(),
-				Enabled:  true,
-			},
-		},
-		GrafanaDashboards: []basereconciler.GrafanaDashboard{
-			{
-				Template: gen.GrafanaDashboard(),
-				Enabled:  !instance.Spec.GrafanaDashboard.IsDeactivated(),
-			},
-		},
-	})
+	// Que resources
+	var que_resources []basereconciler.Resource
+	que_resources, err = r.NewDeploymentWorkload(ctx, instance, r.GetScheme(), &gen.Que)
```

Comment:
Whenever possible the short variable declaration form can be used:

```suggestion
	que_resources, err := r.NewDeploymentWorkload(ctx, instance, r.GetScheme(), &gen.Que)
```

This helps make the code more readable and compact.